src/smt/smt_engine.h

```diff
@@ -341,7 +345,8 @@ class CVC4_PUBLIC SmtEngine {
    * be called when d_logic is updated.
    */
   void setLogicInternal() throw();
-
+  
+  friend class ::CVC4::preprocessing::PreprocessingPassContext;
```

Comment:
Add a TODO to clean up these friend relationship. They do not belong in a public class. Maybe add a github issue and reference that.

Reply:
Done.